### ISSUE TITLE
Which DASH-IF IOP feature(s) should we add to FFmeg? 

### ISSUE THREAD

**Issue by irajs** [Dash-Industry-Forum/FFmpeg#1]

**Comment by nicoweil** [comment]
I think we need to focus on what is realistic from a FFmpeg architecture perspective, which means leveraging signaling in the  source signal to get specific output features. Trying to leverage one single input format is better as it will prevent to use multiple workflows for multiple output features. I would prioritize RTP/TS over any other input formats as it doesn't require an intermediary server to feed FFmpeg (like RTMP does require) and because this is the broadcast industry standard.

1. Multiperiod support, for ad insertion experimentations in low latency mode. RTP with SCTE markers works great for that.

2. Timed Metadata Track support. RTP with SCTE markers would probably work also, but we would need to choose one segmentation type as there is no specific type for such messages in the SCTE-35 spec. I would pick the Content Identification segmentation type as it is kinda generic. A backup plan, if RTP/SCTE is too complex for this purpose, is to use a RTMP source with AMF metadata / cue points.

3. IMSC1 Text profile subtitles support. RTP/DVB-Text would probably be the best source approach.

4. Trickplay support. A RTP source works also great here, for thumbnails extraction (not sure we want to output a low framerate track in the video adaptation set)

**Comment by wilaw** [comment]
I like this list but would also prioritize Resync elements (type = 2) as I think that could lead to better performing lower latency pretty quickly and we can’t build a player for these until we have an encoder/packager that creates them.

Cheers
Will

**Comment by nicoweil** [comment]
Correct, I had forgotten the Resync type 2, this is truly very important at this stage !

**Comment by irajs** [comment]
I suggest we separate the input support vs the output support, like multiperiod support. 

I have a question regarding the input support. Where does the DASH Ingest fit here if any?

**Comment by nicoweil** [comment]
DASH Ingest is an output from the FFmpeg perspective.

We can also request the implementation of DASH-IF ingest interfaces in Nginx, in order to propose the end-to-end solution as open source. That would make sense.

**Comment by irajs** [comment]
That would be good. 

Overall, similar to the overall architecture diagram we have in our IOP document, it would be good we have an architecture diagram for our testbed (I think at one point you had a blueprint), with the list of currently supported relevant formats/protocols at each interface.  Then we can create a shortlist of features for each module, trying to align the necessary support for each feature end-to-end for the relevant components.
